gradle/java/jar-manifest.gradle

```diff
@@ -61,8 +61,9 @@ subprojects {
                             "Specification-Version" : project.baseVersion,
                             "Specification-Title"   : title,
 
-                            "X-Compile-Source-JDK"  : "${project.sourceCompatibility}",
-                            "X-Compile-Target-JDK"  : "${project.targetCompatibility}",
+                            // Evaluate these properties lazily so that the defaults are applied properly.
+                            "X-Compile-Source-JDK"  : "${-> project.sourceCompatibility}",
```

Comment:
I'm not familiar with this syntax and am having trouble searching for it. Can you help me find a reference?

Reply:
I think you're familiar with it, you just don't know it. :) I recall I head-scratched over the very same thing... It's a gstring (lazily evaluated string) with a snippet of gradle code which happens to be a parameterless closure. The jar plugin is actually better now and supports lazy providers... So this could be modified entirely. I just copied the solution that worked in Lucene.

Reply:
This doc explains the difference, btw.
https://docs.groovy-lang.org/latest/html/documentation/#_special_case_of_interpolating_closure_expressions